Source/MoreDotNet.Test/Extensions/Numeric/RomanNumeralExtensions/ParseRomanNumeralTests.cs

```diff
@@ -0,0 +1,321 @@
+            var input = "D";
+            var actual = input.ParseRomanNumeral();
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void ParseRomanNumeral_GivenDI_ShouldReturn501()
+        {
+            var expected = 501;
+            var input = "DI";
+            var actual = input.ParseRomanNumeral();
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void ParseRomanNumeral_GivenDCXLIX_ShouldReturn649()
+        {
+            var expected = 649;
+            var input = "DCXLIX";
+            var actual = input.ParseRomanNumeral();
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void ParseRomanNumeral_GivenDCCXCVIII_ShouldReturn798()
+        {
+            var expected = 798;
+            var input = "DCCXCVIII";
+            var actual = input.ParseRomanNumeral();
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void ParseRomanNumeral_GivenDCCCXCI_ShouldReturn891()
+        {
+            var expected = 891;
+            var input = "DCCCXCI";
+            var actual = input.ParseRomanNumeral();
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void ParseRomanNumeral_GivenM_ShouldReturn1000()
+        {
+            var expected = 1000;
+            var input = "M";
+            var actual = input.ParseRomanNumeral();
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void ParseRomanNumeral_GivenMIV_ShouldReturn1004()
+        {
+            var expected = 1004;
+            var input = "MIV";
+            var actual = input.ParseRomanNumeral();
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void ParseRomanNumeral_GivenMVI_ShouldReturn1006()
+        {
+            var expected = 1006;
+            var input = "MVI";
+            var actual = input.ParseRomanNumeral();
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void ParseRomanNumeral_GivenMXXIII_ShouldReturn1023()
+        {
+            var expected = 1023;
+            var input = "MXXIII";
+            var actual = input.ParseRomanNumeral();
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void ParseRomanNumeral_GivenMMXIV_ShouldReturn2014()
+        {
+            var expected = 2014;
+            var input = "MMXIV";
+            var actual = input.ParseRomanNumeral();
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void ParseRomanNumeral_GivenMMMCMXCIX_ShouldReturn3999()
+        {
+            var expected = 3999;
+            var input = "MMMCMXCIX";
+            var actual = input.ParseRomanNumeral();
+            Assert.Equal(expected, actual);
+        }
+
+        [Fact]
+        public void ParseRomanNumeral_GivenEmptyString_ShouldThrowArgumentException()
```

Comment:
Same for the exception cases - merge into one.
Additional note:
You are missing some test cases:
- What happens if the input is null?
- What happens if the input is all empty spaces?